### PR TITLE
Add Access-Control cluster to all example apps using ZAP

### LINKED ISSUE #14461: Add access control cluster to any example apps that don't have it yet
It's in some example apps, will need to be added to the remaining ones.

### PR DESCRIPTION
**Pull request by harsha-rajendran**
#### Problem
https://github.com/project-chip/connectedhomeip/issues/14461
Add Access-Control cluster to all remaining example apps

#### Change overview
Used ZAP GUI to add access-control cluster (server) to all example apps. Additionally, the Bridge, door-lock, lighting, and OTA-Provider apps also has the Access-control client cluster enabled.

#### Testing
- Ensured local build and tests all ran successfully

### CODE REVIEW

**Comment by hrajendran-samsung** [comment]
@mlepage-google here's the #14461 PR. Would appreciate any comments. TIA

File: examples/light-switch-app/light-switch-common/light-switch-app.matter [hunk 1]
```diff
@@ -595,6 +618,35 @@ client cluster Groups = 4 {
   readonly attribute bitmap8 nameSupport = 0;
   readonly global attribute attrib_id attributeList[] = 65531;
   readonly global attribute int16u clusterRevision = 65533;
+
+  request struct AddGroupRequest {
```
Comment (mlepage-google) [inline examples/light-switch-app/light-switch-common/light-switch-app.matter:622]: Is this meant to be changed/added also?

File: examples/light-switch-app/light-switch-common/light-switch-app.matter [hunk 2]
```diff
@@ -1093,6 +1149,91 @@ client cluster Scenes = 5 {
   readonly attribute bitmap8 nameSupport = 4;
   readonly global attribute attrib_id attributeList[] = 65531;
   readonly global attribute int16u clusterRevision = 65533;
+
+  request struct AddSceneRequest {
```
Comment (mlepage-google) [inline examples/light-switch-app/light-switch-common/light-switch-app.matter:1153]: And this scene stuff?

**Review by mlepage-google — COMMENTED** [review]
Not sure if all the zap changes are intended, or if it's just conflicts or indeterminancy?

File: examples/light-switch-app/light-switch-common/light-switch-app.matter (hunk 2 above)
Comment (hrajendran-samsung) [reply examples/light-switch-app/light-switch-common/light-switch-app.matter:1153]: No, I did not add these (AddSceneRequest and AddGroupRequest) on purpose. Is it possible that ZAP made this decision based on the Access Control Cluster being turned ON?

Comment (mlepage-google) [reply examples/light-switch-app/light-switch-common/light-switch-app.matter:1153]: Not that I know of, but I am not a ZAP expert. Maybe those apps just haven't been ZAP regenerated lately? (But I thought CI checked for that?)

Comment (hrajendran-samsung) [reply examples/light-switch-app/light-switch-common/light-switch-app.matter]: ZAP Regeneration post-rebase fixed it.

Comment (hrajendran-samsung) [reply examples/light-switch-app/light-switch-common/light-switch-app.matter]: Seems pretty finicky- every other regen adds it back, while the other removes it.

**Review by bzbarsky-apple — APPROVED** [review]

**Review by mlepage-google — APPROVED** [review]
Looks better now, just keep an eye on the zap when rebasing etc. thanks!

**Review by msandstedt — APPROVED** [review]

**Comment by mlepage-google** [comment]
Verified cluster is present (on Linux) and functioning in these apps:
- door lock
- lighting
- thermostat
- tv

Could not commission the bridge app for some reason, so couldn't check it.

**Comment by mlepage-google** [comment]
Looks like all directories in examples had a zap file updated, except:

`all-clusters-app build-overrides chip-tool chip-tool-darwin common ipv6only-app minimal-mdns persistent-storage pigweed-app placeholder platform shell`

all-clusters-app was already done, and the rest look like real exceptions (i.e. not apps) to me. None of them have zap files in this list, except placeholder (which seems to be some sort of template app):

```
$ find . -name '*.zap'
./temperature-measurement-app/esp32/main/temperature-measurement.zap
./bridge-app/bridge-common/bridge-app.zap
./all-clusters-app/all-clusters-common/all-clusters-app.zap
./window-app/common/window-app.zap
./log-source-app/log-source-common/log-source-app.zap
./pump-controller-app/pump-controller-common/pump-controller-app.zap
./ota-requestor-app/ota-requestor-common/ota-requestor-app.zap
./light-switch-app/light-switch-common/light-switch-app.zap
./lighting-app/lighting-common/lighting-app.zap
./thermostat/thermostat-common/thermostat.zap
./lock-app/lock-common/lock-app.zap
./ota-provider-app/ota-provider-common/ota-provider-app.zap
./tv-casting-app/tv-casting-common/tv-casting-app.zap
./pump-app/pump-common/pump-app.zap
./placeholder/linux/apps/app1/config.zap
./placeholder/linux/apps/app2/config.zap
./door-lock-app/door-lock-common/door-lock-app.zap
./tv-app/tv-common/tv-app.zap
```

**Pull request merged**